formulaic/transforms/basis_spline.py

```diff
@@ -98,18 +98,21 @@ def basis_spline(  # pylint: disable=dangerous-default-value  # always replaced
     if df is not None and knots is not None:
         raise ValueError("You cannot specify both `df` and `knots`.")
 
+    # Ensure ndarray to avoid conversion overhead
+    x = numpy.asarray(x)
+
     if "lower_bound" in _state:
-        lower_bound = _state["lower_bound"]
+        lower_bound = float(_state["lower_bound"])
     else:
         lower_bound = _state["lower_bound"] = (
-            numpy.min(x) if lower_bound is None else lower_bound
+            float(numpy.nanmin(x)) if lower_bound is None else float(lower_bound)
```

Comment:
nit: Take the float out of the parentheses so we can apply it just once. (also below)

Reply:
I refactored this to avoid 1 level of conditional nesting.